File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
import math

def is_prime(n):
    if n == 0 or n == 1:
        return False
    for i in range(2, int(math.sqrt(n))+1):
        if (n % i == 0):
            return False
    return True

def solution(numbers):
    answer = []
    papers = []
    for i in range(1, len(numbers)+1):
        papers += list(permutations(list(numbers), i))
    
    for paper in papers:
        paper = int(''.join(paper))
        if is_prime(paper):
            answer.append(paper)
    
    answer = list(set(answer))
    return len(answer)

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    int answer = 0;
    Set<Integer> s = new HashSet<>();
    boolean[] visited;
    
    public int solution(String numbers) {
        String[] nums = numbers.split("");
        visited = new boolean[nums.length];
        combination(nums, "");
        
        for (int paper : s) {
            if (checkPrime(paper))
                answer++;
        }
        
        return answer;
    }
    
    private void combination(String[] nums, String curr) {
        if (!curr.equals(""))
            s.add(Integer.parseInt(curr));
        
        for (int i = 0; i < nums.length; i++) {
            if (!visited[i]) {
                visited[i] = true;
                combination(nums, curr+nums[i]);
                visited[i] = false;
            }
        }
    }
    
    private boolean checkPrime(int n) {
        if (n == 0 || n == 1)
            return false;
        for (int i = 2; i <= (int)Math.sqrt(n); i++) {
            if (n % i == 0)
                return false;
        }
        return true;
    }
}